### PR TITLE
chore(middlewares/csp): set `useDefaults` to `true` by default

### LINKED ISSUE #314: Content-Security-Policy: set useDefaults to true by default
In the next major version of Helmet, we should set CSP's `useDefaults` option to `true` by default.

This involves:

1. Updating tests
1. Updating the source code to change the default (should be a very short change)
1. Updating the documentation
1. Adding an entry to the top-level Helmet changelog and the CSP changelog

This change should be made against the `v5.x` branch.

### PR DESCRIPTION
**Pull request by Fdawgs**
Closes #314

@EvanHahn not 100% on if this constitutes a major version bump?

### CODE REVIEW

File: test/content-security-policy.test.ts [hunk 1]
```diff
@@ -372,9 +372,7 @@ describe("Content-Security-Policy middleware", () => {
       contentSecurityPolicy({
         directives: {},
       });
-    }).toThrow(
-      /^Content-Security-Policy has no directives. Either set some or disable the header$/
-    );
+    }).not.toThrow();
```
Comment (EvanHahn) [inline test/content-security-policy.test.ts:375]: I think this test should continue to `throw`, which means we'll need to set `useDefaults: false`.

File: middlewares/content-security-policy/README.md [hunk 2]
```diff
@@ -39,7 +39,7 @@ If no directives are supplied, the following policy is set (whitespace added for
     style-src 'self' https: 'unsafe-inline';
     upgrade-insecure-requests
 
-You can use this default with the `useDefaults` option. `useDefaults` is `false` by default, but will be `true` in the next major version of this module.
+You can use this default with the `useDefaults` option. `useDefaults` is `true` by default.
```
Comment (EvanHahn) [inline middlewares/content-security-policy/README.md:42]: Could you also make this change in the top-level `README.md`?

**Review by EvanHahn — CHANGES_REQUESTED** [review]
Thanks for doing this!

1. Could we add at least one test that demonstrates overriding the defaults when `useDefaults` is completely unset? That test should fail on the old code and succeed in the new code.
1. Could you add the following to the top-level Helmet changelog:

   ```markdown
   ### Changed
   
   - `helmet.contentSecurityPolicy`: `useDefaults` option now defaults to `true`
   ```

File: middlewares/content-security-policy/CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,11 @@
 # Changelog
 
+## 4.0.0 - 2021-05-05
```
[the remark this answers is not in the document]
Comment (Fdawgs) [reply middlewares/content-security-policy/CHANGELOG.md]: Amended in https://github.com/helmetjs/helmet/pull/316/commits/102234273c28ecdda6d27c8487214f2d1493411c

File: middlewares/content-security-policy/README.md (hunk 2 above)
Comment (Fdawgs) [reply middlewares/content-security-policy/README.md:42]: Amended in https://github.com/helmetjs/helmet/pull/316/commits/102234273c28ecdda6d27c8487214f2d1493411c

File: middlewares/content-security-policy/package-overrides.json [hunk 4]
```diff
@@ -6,6 +6,6 @@
     "Ryan Cannon <ryan@ryancannon.com> (https://ryancannon.com)"
   ],
   "description": "Content Security Policy middleware",
-  "version": "3.4.0",
+  "version": "4.0.0",
```
[the remark this answers is not in the document]
Comment (Fdawgs) [reply middlewares/content-security-policy/package-overrides.json]: Fixed in https://github.com/helmetjs/helmet/pull/316/commits/99dcf4c513ae71b5bb657db2505cd9ec4e1d0a68

File: test/content-security-policy.test.ts (hunk 1 above)
Comment (Fdawgs) [reply test/content-security-policy.test.ts]: Fixed in https://github.com/helmetjs/helmet/pull/316/commits/2c497fbb1dd47187852aee1dc51008d40958ade4

**Comment by Fdawgs** [comment]
> 
> 
> Thanks for doing this!
> 
>     1. Could we add at least one test that demonstrates overriding the defaults when `useDefaults` is completely unset? That test should fail on the old code and succeed in the new code.
> 
>     2. Could you add the following to the top-level Helmet changelog:
>        ```gfm
>        ### Changed
>        
>        - `helmet.contentSecurityPolicy`: `useDefaults` option now defaults to `true`
>        ```

Sure @EvanHahn, no problem!

1. Added in https://github.com/helmetjs/helmet/pull/316/commits/b32a28779c2fcb30f78e90abd1c0e6de6e7339c1
2. Added in https://github.com/helmetjs/helmet/pull/316/commits/102234273c28ecdda6d27c8487214f2d1493411c

**Comment by EvanHahn** [comment]
Thanks for doing this! I'll take another look soon.

**Comment by EvanHahn** [comment]
I haven't forgotten about this, but it's low on my list because I don't intend to release Helmet v5 soon. Still hoping to review this in the next couple of weeks, though!

**Review by EvanHahn — APPROVED** [review]
Thanks!

**Comment by EvanHahn** [comment]
I made a mistake when rebasing the `v5.x` branch, so I "redid" this commit in 16f883fdf50ab473f746f0032f3ddb3caa44b85b.

**Pull request merged**